app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php

```diff
@@ -115,6 +115,14 @@ public function execute()
             $response->setError(true);
             $response->setProductAttribute($attribute->toArray());
         }
+        if(in_array($attributeCode, \Magento\Catalog\Model\ResourceModel\Eav\Attribute::DEFAULT_ATTRIBUTE)){
+    		$message = 	__('Cannot create attribute (%1) as it is already used in catalog',$attributeCode);
+    		$this->setMessageToResponse($response, [$message]);
+    		$response->setError(true);
+    		$response->setProductAttribute($attribute->toArray());
```

Comment:
Please add a proper check into

```
if ($attribute->getId() && !$attributeId) {
            $message = strlen($this->getRequest()->getParam('attribute_code'))
                ? __('An attribute with this code already exists.')
                : __('An attribute with the same code (%1) already exists.', $attributeCode);
```

condition, do not introduce a constant - just compare with `'product_type'` string literal.

Reply:
so instead of introducing a constant...i should compare the string..if string is similar then throw msg right?

Reply:
@GovindaSharma exactly, no need to overcomplicate things. When we need more than one such forbidden not-really-an-attribute, we may think about array.

Reply:
hi @orlangur 

"Please fill all description fields according to template and apply code changes, then squash it into single commit and force push."

this was your feedback..i did not understand it.can you please tell me give any idea what changes i need to apply in code?

Reply:
@GovindaSharma check this: https://github.com/magento/magento2/pull/19408/files#r236223352

You need to add new check into existing `if` instead of introducing a new one.

Reply:
@GovindaSharma still not fixed